src/App.tsx

```diff
@@ -14,11 +14,110 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  goods: string[]
+  isVisible: boolean;
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: goodsFromServer,
+    isVisible: false,
+  };
+
+  showGoods = () => {
+    this.setState(state => ({
+      isVisible: !state.isVisible,
+    }));
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      goods: [...state.goods].reverse(),
+    }));
+  };
+
+  sortByLength = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort((goodOne, goodTwo) => goodOne.length - goodTwo.length),
+    }));
+  };
+
+  sortByAplhabet = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort((goodOne, goodTwo) => goodOne.localeCompare(goodTwo)),
+    }));
+  };
+
+  reset = () => {
+    this.setState({
+      goods: [...goodsFromServer],
+    });
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <h1 className="App__title">List of Goods</h1>
```

Comment:
You have lots of classes, but no styles associated with them. Either remove unused classNames or add s styles.